### PR TITLE
[3713] Add float type for controls to be able to use decimals in proptypes

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/3713

### CODE REVIEW

File: static-assets/components/cstudio-admin/mods/content-type-propsheet/float.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    this.valueEl = valueEl;
+  },
+
+  getValue: function() {
+    return this.valueEl.value;
+  },
+
+  isNumberKey: function(charCode) {
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/content-type-propsheet/float.js:66]: Please annotate the code with the keys that you're checking and comments around what you're doing. 

Also, looks like these aren't char codes but key codes.

Is this considering numeric keypad?

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-forms/controls/numeric-input.js [hunk 2]
```diff
@@ -106,11 +106,12 @@ YAHOO.extend(CStudioForms.Controls.numericInput, CStudioForms.CStudioFormField,
 	 * @param el element
 	 */
 	count: function(evt, countEl, el) {
-		var max = this.getAttribute("max") ,
-			min = this.getAttribute("min");
-		if(max && parseInt(this.value) > max){
+	  var el = (el) ? el : this,
+        max = this.getAttribute("max") ,
+			  min = this.getAttribute("min");
```
Comment (rart) [inline static-assets/components/cstudio-forms/controls/numeric-input.js]: Where is getAttribute declared?

Comment (jvega190) [reply static-assets/components/cstudio-forms/controls/numeric-input.js]: @rart yeah sorry, I need https://github.com/craftercms/studio-ui/pull/1475 to be merged so I can update this branch

File: static-assets/components/cstudio-admin/mods/content-type-propsheet/float.js (hunk 1 above)
Comment (jvega190) [reply static-assets/components/cstudio-admin/mods/content-type-propsheet/float.js:66]: yes it is considering numeric keypad, will comment

**Review by rart — APPROVED** [review]

**Pull request merged**